DataFrame.plot error when both 'color' and 'style' arguments are passed (GH9671)

davidbrochart opened the pull request:
closes #9671

`self.style` can be a list of strings or a string. If it is a string, we make a list containing that string, so that we only have to deal with a list of strings. We check every element in that list and raise an error if it has a color symbol.
The test makes sure that calling `plot()` and passing both `color` and `style` arguments is allowed if there is no color symbol in the style string(s), and that passing both `color` and `style` arguments is not allowed if there is a color symbol in the style string(s). This test is done for a dataframe and for a series.

Commit:
Fixed bug #9671 where 'DataFrame.plot()' raised an error when both 'color' and 'style' keywords were passed and there was no color symbol in the style strings (this should be allowed)

Commit:
Fix for issue #9671

davidbrochart committed:
Added test for bug fix of issue #9671

davidbrochart committed:
Merge remote-tracking branch 'upstream/master' into HEAD

TomAugspurger commented:
Thanks. In the future, you don't need to open up a new PR. You can push changes to your original pull request and github will pick them up.

TomAugspurger commented on pandas/tests/test_graphics.py:
```diff
@@ -991,6 +991,18 @@ def test_plot(self):
         df = DataFrame({'x': [1, 2], 'y': [3, 4]})
         with tm.assertRaises(TypeError):
             df.plot(kind='line', blarg=True)
+        try:
```
Instead of wrapping the code in `try` / `except` plots, just call the function.

`ax = df.plot(color=['red', 'black'], style=['-', '--'])` and then make sure the color and styles match what's expected.
You can get the actual linestyle with `result = [line.get_linestyle() for line in ax.lines]`
Then compare that to the expected with `self.assertEqual(result, ['-', '--'])`

TomAugspurger commented on pandas/tests/test_graphics.py:
```diff
@@ -991,6 +991,18 @@ def test_plot(self):
+        except:
+            self.fail("Calling 'plot()' on a dataframe/series and passing both 'color' and 'style' arguments should be allowed if there is no color symbol in the style string(s)")
+        try:
+            df.plot(color = ['red', 'black'], style = ['k-', 'r--'])
```
This one should raise and exception, so you can use the `tm.assertRaises` context manager

```python
with tm.assertRaises(ValueError):
     df.plot(color = ['red', 'black'], style = ['k-', 'r--'])
```

so that the test passes if a ValueError is raised, and fails if it doesn't.

davidbrochart committed:
Made a separate test called test_color_and_style_arguments, made changes as suggested by TomAugspurger

davidbrochart commented:
Thanks for your comments Tom. I made the changes you suggested.

dsm054 committed:
BUG: DataFrame.equals should not care about block order (GH #9330)

tomazberisa committed:
BUG: Fix for #9764

Values from range [1e-7, 5e-7] (for display.precision=7) not displaying 0 anymore

tomazberisa committed:
TST: Test for #9764 fix

tomazberisa committed:
TST: Test saves and restore context (#9764)

tomazberisa committed:
CLN: Test code cleanup (#9764)

tomazberisa committed:
DOC: Update whatsnew for 0.16.1 (#9764)

Commit:
Merge pull request #9812 from sinhrks/sec_legend

BUG: secondary_y may not show legend properly

shoyer committed:
Merge pull request #9806 from tomazberisa/df-display-bug

BUG: format small floats correctly (GH9764)

TomAugspurger commented:
@davidbrochart could you rebase on master and add a release note in `doc/source/whatsnew/v0.16.1.txt`? Thanks.

davidbrochart committed:
DOC: Update whatsnew for 0.16.1 (#9671)

davidbrochart committed:
Merge branch 'plot_style_fix_and_test' of https://github.com/davidbrochart/pandas into plot_style_fix_and_test

davidbrochart commented:
@TomAugspurger I rebased on master and added a release note. Hope this is fine.

TomAugspurger commented:
Merged via https://github.com/pydata/pandas/commit/f00d6bba5a7ee42569b9d6f7848962fd04a3360d

Thanks @davidbrochart
